doc/engine.md

```diff
@@ -0,0 +1,215 @@
+than in any other commonly available game engine.
+
+After picking an art style based on the available features, an artist should
+not be forced to deal with new features incompatible with their art.
+New visual features should always accomodate all visual styles, providing
+an opt-out for developers who have no use for them. This opt-out must
+come in the form of scripting API provisions.
+It is unacceptable for a creator or server admin to have to ask the user
+to configure the client in a certain way.
+
+A content creator's restrictions on presentation take precedence over
+personal preference. It should not be easy for a user to unknowingly
+configure their client in a way where they may get a sub-par experience,
+and possibly even blame the content creator for it.
+Understand that some types of artwork and some rendering features do not mix.
+A model made and textured for an unlit renderer is unlikely to look
+better with gouraud shading or shadows, it's more likely to look worse.
+
+Heed the words of one dissatisfied user: (Issue #10307, worth a read)
+
+> When people enable additional graphics features, they expect better graphics, not worse
+
+An implementation of a visual feature should aim for a balanced
+quality and performance profile that fits a general use case and works
+on a wide range of devices.
+
+Visual features that do not work on all supported devices may be regarded
+as low priority. (This does not apply to pure optimizations.)
+
+### 3.1. Client visual settings
+
+While the client may toggle various visual features for the sake
+of personal preference and performance, games should not be
+forced to utilize or anticipate those that significantly
+change the way content is being rendered.
+
+Examples of visually neutral settings:
+* Screen resolution
+* Maximum FPS
+* Vertical sync
+* Anti-aliasing
+* Texture filtering
+* Stereoscopy
+
+The above settings affect the image only in minor ways,
+and are safe to apply to just about any content.
+
+
+Examples of non-neutral features that should be API gated:
+* Direct lighting
+* Realtime shadows
+* Bump-mapping
+* FOV changes
+
+Above features affect visuals or gameplay in significant ways,
+requiring some degree of gating in the scripting API.
+It's best if the language in the config descriptions is carefully
+chosen to use words such as "allow" rather than "enable", to
+avoid confusion.
+
+It does not suffice to mark a feature as "experimental" or have an explicit
+"force" setting - assume that the user will enable everything you put
+in front of them, even to their detriment. Don't expect every user to know
+what the options actually do.
+
+A third class of visual settings would be speedhacks, such as:
+* Opaque water
+* Opaque leaves
+* No waving
+
+With speedhacks, there is a clear understanding that they are sacrificing
+quality for performance, and possibly breaking things in the process.
+In future GUI redesigns they should probably be labeled more clearly as such.
```

Comment:
Since this entire document sets some guidelines existing feature don't necessarily conform to this can just say:
"These features should be clearly labeled as such in the GUI."

Reply:
Right, I guess it's fine to document how things should be and fix what doesn't comply.